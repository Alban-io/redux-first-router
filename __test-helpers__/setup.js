import createHistory from 'history/createMemoryHistory'
import connectRoutes from '../src/connectRoutes'


export default (path = '/', options = { title: 'title', location: 'location' }) => {
  const routesMap = {
    FIRST: '/first',
    SECOND: '/second/:param',
  }

  const history = createHistory({
    initialEntries: [path],
    initialIndex: 0,
    keyLength: 6,
  })

  return connectRoutes(history, routesMap, options)
}
