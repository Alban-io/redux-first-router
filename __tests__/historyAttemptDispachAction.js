import setup, { setupAll } from '../__test-helpers__/setup'

it('historyAttemptDispatchAction: dispatches action matching pathname when history location changes', () => {
  const dispatch = jest.fn()
  const store = { dispatch }
  const historyLocation = { pathname: '/second/foo' }
  const { _historyAttemptDispatchAction } = setup()

  _historyAttemptDispatchAction(store, historyLocation, 'pop')

  const action = dispatch.mock.calls[0][0] /*? */

  expect(action.type).toEqual('SECOND')
  expect(action.payload).toEqual({ param: 'foo' })
  expect(action.meta.location.kind).toEqual('pop')

  expect(action).toMatchSnapshot()
})

it('historyAttemptDispatchAction: does not dispatch if pathname is the same (i.e. was handled by middleware already)', () => {
  const dispatch = jest.fn()
  const store = { dispatch }
  const historyLocation = { pathname: '/second/foo' }
  const { _historyAttemptDispatchAction } = setup()

  _historyAttemptDispatchAction({ dispatch: jest.fn() }, historyLocation, 'pop')
  _historyAttemptDispatchAction(store, historyLocation, 'pop')

  // insure multiple dispatches are prevented for the same action/pathname
  // so that middleware and history listener don't double dispatch
  expect(dispatch.mock.calls).toEqual([])
})
