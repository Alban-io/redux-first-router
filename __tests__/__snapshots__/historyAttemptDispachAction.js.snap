// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`historyAttemptDispatchAction: dispatches action matching pathname when history location changes 1`] = `
Object {
  "kind": "pop",
  "meta": Object {
    "location": Object {
      "current": Object {
        "kind": "pop",
        "pathname": "/second/foo",
        "payload": Object {
          "param": "foo",
        },
        "type": "SECOND",
      },
      "history": undefined,
      "kind": "pop",
      "prev": Object {
        "kind": "",
        "pathname": "",
        "payload": Object {},
        "type": "",
      },
    },
  },
  "payload": Object {
    "param": "foo",
  },
  "type": "SECOND",
}
`;
